### DISCUSSION TITLE
Cache `when` and `unless`

### DISCUSSION THREAD

**Discussion by devajmeireles** [laravel/framework#48255] [Ideas]
Since if I suggest the PR it will be disapproved 😂 , I'm posting it here for those who want to do it:

It would be awesome if `Cache` facade supported the `when` and `unless` methods, to provide us with something like this:

```php
Cache::when(/* ... */, fn (Cache $cache) => $cache->rememberForever($key, fn() => $post))
```

In theory is just to add the `Conditionable` trait to the `Cache` repository.

**Comment by Angel5a** [comment]
Conditionable returns $this. I don't see any benefits. For one-liner fun there is a way to avoid of using if in less keystrokes:
```php
/* ... */ && Cache::rememberForever($key, fn() => $post))
```
